deepspeed/launcher/runner.py

```diff
@@ -432,7 +434,7 @@ def main(args=None):
     if multi_node_exec and not args.no_ssh_check:
         first_host = list(active_resources.keys())[0]
         try:
-            subprocess.check_call(f'ssh -o PasswordAuthentication=no {first_host} hostname',
+            subprocess.check_call(f'ssh -o PasswordAuthentication=no -p {args.port} {first_host} hostname',
```

Comment:
I think we only want to pass `-p {args.port}` iff --ssh_port is set. In many cases we have users setting up a [ssh config](https://phoenixnap.com/kb/ssh-config) that sets specific port/connection parameters. If we force passing `-p 22` as the default then this will break a ssh config host that may have a port that is not 22.

Reply:
Thank you for your feedback. I've added an if statement to make sure the `-p` is only added on when `args.ssh_port` is specified.